src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -321,12 +321,12 @@ AccountRootsNotDeleted::finalize(
     ReadView const&,
     beast::Journal const& j)
 {
-    // AMM account root can be deleted as the result of AMM withdraw
+    // AMM account root can be deleted as the result of AMM withdraw/delete
     // transaction when the total AMM LP Tokens balance goes to 0.
-    // Not every AMM withdraw deletes the AMM account, accountsDeleted_
-    // is set if it is deleted.
+    // A successful AccountDelete or AMMDelete MUST delete exactly
+    // one account root.
     if ((tx.getTxnType() == ttACCOUNT_DELETE ||
-         (tx.getTxnType() == ttAMM_WITHDRAW && accountsDeleted_ == 1)) &&
+         tx.getTxnType() == ttAMM_DELETE) &&
```

Comment:
This should be tx.getTxnType() == AMM_WITHDRAW || tx.getTxnType() == AMM_DELETE right?